pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go

```diff
@@ -69,66 +66,107 @@ func (db *AzureSqlDbManager) Ensure(ctx context.Context, obj runtime.Object, opt
+				}
+				instance.Status.Message = fmt.Sprintf("Azure DB long-term retention policy error: %s", errhelp.StripErrorIDs(err))
+				azerr := errhelp.NewAzureErrorAzureError(err)
+				if helpers.ContainsString(failureErrors, azerr.Type) {
+					instance.Status.Provisioning = false
+					instance.Status.Provisioned = false
+					instance.Status.FailedProvisioning = true
+					return true, nil
+				} else {
+					return false, err
+				}
 			}
-			instance.Status.Message = fmt.Sprintf("Azure DB long-term retention policy error: %s", errhelp.StripErrorIDs(err))
+
+			// db exists, we have successfully provisioned everything
+			instance.Status.Provisioning = false
+			instance.Status.Provisioned = true
+			instance.Status.FailedProvisioning = false
+			instance.Status.State = string(dbGet.Status)
+			instance.Status.Message = resourcemanager.SuccessMsg
+			instance.Status.ResourceId = *dbGet.ID
+			return true, nil
+		} else {
+			instance.Status.Message = fmt.Sprintf("AzureSqlDb Get error %s", err.Error())
 			azerr := errhelp.NewAzureErrorAzureError(err)
-			if helpers.ContainsString(failureErrors, azerr.Type) {
+			requeuErrors := []string{
+				errhelp.ParentNotFoundErrorCode,
+				errhelp.ResourceGroupNotFoundErrorCode,
+			}
+			if helpers.ContainsString(requeuErrors, azerr.Type) {
 				instance.Status.Provisioning = false
-				instance.Status.Provisioned = false
-				instance.Status.FailedProvisioning = true
-				return true, nil
-			} else {
-				return false, err
+				return false, nil
 			}
 		}
-
-		// db exists, we have successfully provisioned everything
-		instance.Status.Provisioning = false
-		instance.Status.Provisioned = true
-		instance.Status.FailedProvisioning = false
-		instance.Status.State = string(dbGet.Status)
-		instance.Status.Message = resourcemanager.SuccessMsg
-		instance.Status.ResourceId = *dbGet.ID
-		return true, nil
-	}
-	instance.Status.Message = fmt.Sprintf("AzureSqlDb Get error %s", err.Error())
-	azerr := errhelp.NewAzureErrorAzureError(err)
-	requeuErrors := []string{
-		errhelp.ParentNotFoundErrorCode,
-		errhelp.ResourceGroupNotFoundErrorCode,
-	}
-	if helpers.ContainsString(requeuErrors, azerr.Type) {
-		instance.Status.Provisioning = false
-		return false, nil
 	}
 
-	resp, err := db.CreateOrUpdateDB(ctx, groupName, location, server, labels, azureSQLDatabaseProperties)
+	pollingUrl, _, err := db.CreateOrUpdateDB(ctx, groupName, location, server, labels, azureSQLDatabaseProperties)
+
 	if err != nil {
 		instance.Status.Message = err.Error()
 		azerr := errhelp.NewAzureErrorAzureError(err)
 
 		// handle errors
 		// resource request has been sent to ARM
 		if azerr.Type == errhelp.AsyncOpIncompleteError {
+			instance.Status.Message = "Resource request successfully submitted to Azure"
 			instance.Status.Provisioning = true
+			instance.Status.PollingURL = pollingUrl
 			return false, nil
 		}
 
+		if azerr.Type == errhelp.InvalidMaxSizeTierCombination {
+			instance.Status.FailedProvisioning = true
```

Comment:
should we hide these two status fields from modification and instead provide methods that update them correctly?

Reply:
We probably should, yes -- and maybe in the future consider merging some of them (i.e. `Provisioning` and `FailedProvisioning` could be like `ProvisioningStatus` as an enum instead of 2 bools that represent 3 distinct states (since 1 of the 4 is invalid).

Reply:
Probably in a different PR though>

Reply:
sure

Reply:
I'll file an issue for it

Reply:
https://github.com/Azure/azure-service-operator/issues/1243